pkg/eventsapi/retrytransport.go

```diff
@@ -64,12 +64,13 @@ func (r RetryTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 			r.log.Debugf("Successful or non-retryable response.")
 			return resp, err
 		} else if !r.IsRetryable(resp, err) {
-			if err == nil {
-				err = ErrAPIError
+			if resp != nil {
```

Comment:
Adheres to [RoundTripper interface](https://golang.org/pkg/net/http/#RoundTripper) by returning `err == nil` in the event an HTTP response is received.

The calling code in `pkg/eventsapi/eventsapi.go` only attempts to parse the response body if no `err` is returned so this helps there as well.